java/ql/lib/semmle/code/java/dispatch/Unification.qll

```diff
@@ -0,0 +1,136 @@
+/**
+ * Provides a module to check whether two `ParameterizedType`s are unifiable.
```

Comment:
I would probably make this an internal module, by moving inside of an `internal` folder.